.github/workflows/daily_modin_precommit.yml

```diff
@@ -215,6 +215,74 @@ jobs:
             .tox/.coverage
             .tox/coverage.xml
 
+  test-enable-cte-optimization:
+    name: Test Enable CTE Optimization modin-${{ matrix.os.download_name }}-${{ matrix.python-version }}-${{ matrix.cloud-provider }}
+    needs: build
+    runs-on: ${{ matrix.os.image_name }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - image_name: macos-latest
+            download_name: macos  # it includes doctest
```

Comment:
it seems we are using macos for most daily runs, but we have larger machines for linux and windows, shall we use a different one?

Reply:
Now you use ubuntu to run doctests which is new to us. So it may have different behavior than macos. Just a reminder